### PR TITLE
Make worksheet interpretation more robust

### PR DESCRIPTION
**Pull request by kashizui**
No text crashes so far anymore.

Fixes codalab/codalab#1424

@percyliang please review

screen shot 2015-11-24 at 12 23 14 pm

### CODE REVIEW

**Comment by percyliang** [comment]
LGTM
Would be nice to have raw line numbers to the error messages.

**Comment by kashizui** [comment]
Any suggestions for how to get the raw line number?

**Comment by kashizui** [comment]
Merged with new changes and added line numbers.

Formatting is inconsistent but that is the nature of our disjointed calls to the client for now.

@percyliang please review

@pbhatnagar3 please take a quick look too since I touched your maps, from my personal testing looks like the jumping is still working though

screen shot 2015-11-30 at 11 57 11 am

**Comment by pujun-ai** [comment]
I played with this and the jumping seems to be working. LGTM :+1:

File: codalab/client/local_bundle_client.py [hunk 1]
```diff
@@ -957,56 +957,69 @@ def resolve_interpreted_items(self, interpreted_items):
...
+                                row = line.split('\t')
+                                points.append(row)
+                elif mode == 'search':
+                    data = worksheet_util.interpret_search(self, None, data)
+                elif mode == 'wsearch':
+                    data = worksheet_util.interpret_wsearch(self, data)
+                elif mode == 'worksheet':
+                    pass
+                else:
+                    raise UsageError('Invalid display mode: %s' % mode)
```
Comment (percyliang) [inline codalab/client/local_bundle_client.py:1015]: I wish GitHub had a way of doing diffs based on changes in indentation.

File: codalab/lib/worksheet_util.py [hunk 2]
```diff
@@ -683,6 +683,10 @@ def flush_bundles():
         if len(bundle_infos) == 0:
             return
 
+        def raise_usage_error():
```
Comment (percyliang) [inline codalab/lib/worksheet_util.py]: This method was meant for only certain type of usage errors involving
display mode genpath
If you move it up here, then rename to raise_genpath_usage_error()

**Comment by percyliang** [comment]
LGTM again (with comments)

**Pull request merged**